src/Compilers/Core/Portable/Syntax/SeparatedSyntaxList.cs

```diff
@@ -566,7 +566,9 @@ private SyntaxNodeOrToken[] NodesWithSeparators
             get { return _list.ToArray(); }
         }
 
+#pragma warning disable RS0041 // uses oblivious reference types
         public Enumerator GetEnumerator()
+#pragma warning restore RS0041 // uses oblivious reference types
```

Comment:
I'm confused by the RS0041 produced here. There is no `#nullable disable` in this file, and NRT should be enabled. Not sure why there is a diagnostic here.

Reply:
`Enumerator` is a nested type. It's usage here is treated like:
```
#nullable disable
SeparatedSyntaxList<T>
#nullable enable 
    .Enumerator
```

See https://github.com/dotnet/roslyn-analyzers/pull/4681 for some illustrations.
This only happens if `T` is is constrained to a reference type if I recall correctly.

This is a problem with the symbol API which we likely won't be able to fix.